connectors/source/firecrawl.py

```diff
@@ -0,0 +1,312 @@
+    # Validate parameters
+    if not api_key:
+        return {"error": "Firecrawl API key is required. Set FIRECRAWL_API_KEY environment variable."}
+    
+    if not s3_uri:
+        return {"error": "S3 URI is required"}
+    
+    # Make sure S3 URI starts with s3:// and ends with /
+    if not s3_uri.startswith("s3://"):
+        return {"error": "S3 URI must start with 's3://'"}
+    
+    # Ensure URI ends with a slash
+    if not s3_uri.endswith("/"):
+        s3_uri += "/"
+    
+    try:
+        # Initialize the Firecrawl client
+        firecrawl = FirecrawlApp(api_key=api_key)
+        
+        # Prepare crawl parameters
+        params = {
+            "limit": limit,
+            "scrapeOptions": {
+                "formats": ["html"]  # Only use HTML format
+            }
+        }
+        
+
+        
+        # Start the crawl
+        crawl_status = firecrawl.async_crawl_url(url, params=params)
+        
+        # Start background processing in all cases
+        if "id" in crawl_status:
+            crawl_id = crawl_status["id"]
+            # Start background task without waiting for it (fire and forget)
+            asyncio.create_task(
+                wait_for_crawl_completion(ctx, crawl_id, s3_uri)
+            )
+            
+            # Update the response to indicate background processing
+            crawl_status["auto_processing"] = True
+            crawl_status["s3_uri"] = f"{s3_uri}{crawl_id}/"
+            crawl_status["message"] = "Crawl started and will be automatically processed when complete"
+        
+        return crawl_status
+    except Exception as e:
+        return {"error": f"Error starting firecrawl job: {str(e)}"}
+
+
+async def check_crawl_status(
+    ctx: Context,
+    crawl_id: str,
+) -> Dict[str, Any]:
+    """Check the status of an existing Firecrawl crawl job.
+
+    Args:
+        crawl_id: ID of the crawl job to check
+
+    Returns:
+        Dictionary containing the current status of the crawl job
+    """
+    # Get the API key from environment variable
+    api_key = os.getenv("FIRECRAWL_API_KEY")
+    
+    if not api_key:
+        return {"error": "Firecrawl API key is required. Set FIRECRAWL_API_KEY environment variable."}
+    
+    try:
+        # Initialize the Firecrawl client
+        firecrawl = FirecrawlApp(api_key=api_key)
+        
+        # Check the crawl status
+        result = firecrawl.check_crawl_status(crawl_id)
+        
+        # Return a more user-friendly response
+        status_info = {
+            "id": crawl_id,
+            "status": result.get("status", "unknown"),
+            "completed_urls": result.get("completed", 0),
+            "total_urls": result.get("total", 0),
+        }
+        
+            
+        return status_info
+    except Exception as e:
+        return {"error": f"Error checking crawl status: {str(e)}"}
+
+
+def _upload_directory_to_s3(local_dir: str, s3_uri: str) -> Dict[str, Any]:
+    """Upload a directory to S3.
```

Comment:
Just checking again if this should be handled by a destination connector, if it makes sense. It could be that this was a requirement. In which case, it should be ok.